cmd/occollector/app/collector/collector.go

```diff
@@ -0,0 +1,259 @@
+		os.Exit(1)
+	}
+
+	// TODO: (@pjanotti) construct queued processor from config options, for now just to exercise it, wrap one around
+	// the first span processor available.
+	if v.GetBool(queuedProcessorFlg) {
+		spanProcessors[0] = processor.NewQueuedSpanProcessor(spanProcessors[0])
+	}
+
+	// Wraps processors in a single one to be connected to all enabled receivers.
+	spanProcessor := processor.NewMultiSpanProcessor(spanProcessors...)
+
+	receiversCloseFns := createReceivers(spanProcessor)
+	closeFns = append(closeFns, receiversCloseFns...)
+
+	logger.Info("Collector is up and running.")
+
+	<-signalsChannel
+	logger.Info("Starting shutdown...")
+
+	// TODO: orderly shutdown: first receivers, then flushing pipelines giving
+	// senders a chance to send all their data. This may take time, the allowed
+	// time should be part of configuration.
+	for i := len(closeFns) - 1; i > 0; i-- {
+		closeFns[i]()
+	}
+
+	logger.Info("Shutdown complete.")
+}
+
+func createExporters() (doneFns []func(), traceExporters []exporter.TraceExporter) {
+	// TODO: (@pjanotti) this is slightly modified from agent but in the end duplication, need to consolidate style and visibility.
+	parseFns := []struct {
+		name string
+		fn   func([]byte) ([]exporter.TraceExporter, []func() error, error)
+	}{
+		{name: "datadog", fn: exporterparser.DatadogTraceExportersFromYAML},
+		{name: "stackdriver", fn: exporterparser.StackdriverTraceExportersFromYAML},
+		{name: "zipkin", fn: exporterparser.ZipkinExportersFromYAML},
+		{name: "jaeger", fn: exporterparser.JaegerExportersFromYAML},
+		{name: "kafka", fn: exporterparser.KafkaExportersFromYAML},
+	}
+
+	if config == "" {
+		logger.Info("No config file, exporters can be only configured via the file.")
+		return
+	}
+
+	cfgBlob, err := ioutil.ReadFile(config)
+	if err != nil {
+		logger.Fatal("Cannot read config file for exporters", zap.Error(err))
+	}
+
+	for _, cfg := range parseFns {
+		tes, tesDoneFns, err := cfg.fn(cfgBlob)
+		if err != nil {
+			logger.Fatal("Failed to create config for exporter", zap.String("exporter", cfg.name), zap.Error(err))
+		}
+
+		wasEnabled := false
+		for _, te := range tes {
+			if te != nil {
+				wasEnabled = true
+				traceExporters = append(traceExporters, te)
+			}
+		}
+
+		for _, tesDoneFn := range tesDoneFns {
+			if tesDoneFn != nil {
+				wrapperFn := func() {
+					if err := tesDoneFn(); err != nil {
+						logger.Warn("Error when closing exporter", zap.String("exporter", cfg.name), zap.Error(err))
+					}
+				}
+				doneFns = append(doneFns, wrapperFn)
+			}
+		}
+
+		if wasEnabled {
+			logger.Info("Exporter enabled", zap.String("exporter", cfg.name))
+		}
+	}
+
+	return doneFns, traceExporters
+}
+
+func createReceivers(spanProcessor processor.SpanProcessor) (closeFns []func()) {
+	var someReceiverEnabled bool
+	receivers := []struct {
+		name  string
+		runFn func(*zap.Logger, *viper.Viper, processor.SpanProcessor) (func(), error)
+		ok    bool
```

Comment:
shouldn't this be called enabled instead of ok? ok = false seems to imply something went wrong